core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -183,13 +186,16 @@ export class StateObjectFactory {
     });
     if (newState.interaction.defaultOutcome !== null) {
       let defaultOutcome = newState.interaction.defaultOutcome;
-      defaultOutcome.dest = newStateName;
+      defaultOutcome.dest = newStateName as string;
     }
     return newState;
   }
 
+  // Passes newstatename as null untill new question is created it will be null
+  // until fetch from the backend and are not null afterwards.
```

Comment:
I don't understand this, how is state connected to a question. Also, please proofread your comments.

Reply:
Done